File: src/main/java/voidchess/figures/MockFigure.java
package voidchess.figures;

import voidchess.board.BasicChessGameInterface;
import voidchess.board.SimpleChessBoardInterface;
import voidchess.helper.Move;
import voidchess.helper.Position;
import voidchess.image.ImageType;

import java.util.List;

/**
 * @author stephan
 */
class MockFigure extends Figure {
    MockFigure(boolean isWhite, Position position) {
        super(isWhite, position, FigureType.MOCK);
    }

    public boolean isReachable(Position to, BasicChessGameInterface game) {
        return false;
    }

    public void getReachableMoves(BasicChessGameInterface game, List<Move> result) {
    }

    public boolean isSelectable(SimpleChessBoardInterface game) {
        return true;
    }

    public int countReachableMoves(BasicChessGameInterface game) {
        return 0;
    }
}

File: src/main/kotlin/voidchess/figures/MockFigure.kt
package voidchess.figures

import voidchess.board.BasicChessGameInterface
import voidchess.board.SimpleChessBoardInterface
import voidchess.helper.Move
import voidchess.helper.Position
import voidchess.image.ImageType

/**
 * @author stephan
 */
internal class MockFigure(isWhite: Boolean, position: Position) : Figure(isWhite, position, FigureType.MOCK) {

    override fun isReachable(to: Position, game: BasicChessGameInterface) = false
    override fun isSelectable(game: SimpleChessBoardInterface) = true
    override fun getReachableMoves(game: BasicChessGameInterface, result: MutableList<Move>) {}
    override fun countReachableMoves(game: BasicChessGameInterface) = 0
}
